targets/csharp/source/PlayFabSDK/source/PlayFabHttp/PlayFabPollyHttp.cs

```diff
@@ -82,7 +82,7 @@ public new async Task<object> DoPost(string fullPath, object request, Dictionary
             object doPostResult = await CommonResilience
                 .ExecuteAsync(async () =>
                 {
-                    var result = await base.DoPost(fullPath, request, headers) as PlayFabError;
+                    var result = await base.DoPost(fullPath, request, headers) as PlayFabResultCommon;
```

Comment:
I'm not sure this works. When base.DoPost succeeds, the return object is a string. It's only a well-formed object on failure (which is admittedly weird). My local fix was to not cast here, but instead check the type in the retryPolicy and breakerPolicy handlers and only apply policy if the type was PlayFabError.

Reply:
Actually i should update this test because what's happening now is the test also shows we retry until we succeeds, and tries more than 10 times. Ill add an assertion for that check, too.

Reply:
Ok if we dont use the instancename string, then this works. Thanks!!